src/app/file/file.cpp

```diff
@@ -920,6 +920,11 @@ void FileOp::operate(IFileOpProgress* progress)
         setError("Error loading data file: %s\n", ex.what());
       }
     }
+
+    if (hasIncompatibilityError()) {
+      setError("Incompatibility error found: %s.\nDocument will be marked as read-only." \
+               " Please upgrade Aseprite to the latest version to be able to save this document without losing data.", m_incompatibilityError.c_str());
+    }
```

Comment:
Change the text to:
```suggestion
    if (hasIncompatibilityError()) {
      setError("Incompatibility error found:\n\n  %s.\n\n"
               "This file will be opened as read-only."
               " Please upgrade Aseprite to the latest version to be able to save changes without losing data.",
               m_incompatibilityError.c_str());
```
And probably we should add these strings in the en.ini strings.